reconcile/typed_queries/saas_files.py

```diff
@@ -1,26 +1,237 @@
+        ..., alias="promotion"
+    )
+    parameters: Optional[Json] = Field(..., alias="parameters")
+    secret_parameters: Optional[
+        list[SaasResourceTemplateTargetV2_SaasSecretParametersV1]
+    ] = Field(..., alias="secretParameters")
+    upstream: Optional[SaasResourceTemplateTargetUpstreamV1] = Field(
+        ..., alias="upstream"
+    )
+    image: Optional[SaasResourceTemplateTargetImageV1] = Field(..., alias="image")
+    disable: Optional[bool] = Field(..., alias="disable")
+    delete: Optional[bool] = Field(..., alias="delete")
+
+    class Config:
+        # ignore `namespaceSelector` and 'provider' fields from the GQL schema
+        extra = Extra.ignore
+
+
+class SaasResourceTemplate(ConfiguredBaseModel):
+    name: str = Field(..., alias="name")
+    url: str = Field(..., alias="url")
+    path: str = Field(..., alias="path")
+    provider: Optional[str] = Field(..., alias="provider")
+    hash_length: Optional[int] = Field(..., alias="hash_length")
+    parameters: Optional[Json] = Field(..., alias="parameters")
+    secret_parameters: Optional[
+        list[SaasResourceTemplateV2_SaasSecretParametersV1]
+    ] = Field(..., alias="secretParameters")
+    targets: list[SaasResourceTemplateTarget] = Field(..., alias="targets")
+
+
+class SaasFile(ConfiguredBaseModel):
+    path: str = Field(..., alias="path")
+    name: str = Field(..., alias="name")
+    app: AppV1 = Field(..., alias="app")
+    pipelines_provider: Union[PipelinesProviderTektonV1, PipelinesProviderV1] = Field(
+        ..., alias="pipelinesProvider"
+    )
+    deploy_resources: Optional[DeployResourcesV1] = Field(..., alias="deployResources")
+    slack: Optional[SlackOutputV1] = Field(..., alias="slack")
+    managed_resource_types: list[str] = Field(..., alias="managedResourceTypes")
+    takeover: Optional[bool] = Field(..., alias="takeover")
+    deprecated: Optional[bool] = Field(..., alias="deprecated")
+    compare: Optional[bool] = Field(..., alias="compare")
+    timeout: Optional[str] = Field(..., alias="timeout")
+    publish_job_logs: Optional[bool] = Field(..., alias="publishJobLogs")
+    cluster_admin: Optional[bool] = Field(..., alias="clusterAdmin")
+    image_patterns: list[str] = Field(..., alias="imagePatterns")
+    allowed_secret_parameter_paths: Optional[list[str]] = Field(
+        ..., alias="allowedSecretParameterPaths"
+    )
+    use_channel_in_image_tag: Optional[bool] = Field(
+        ..., alias="use_channel_in_image_tag"
+    )
+    authentication: Optional[SaasFileAuthenticationV1] = Field(
+        ..., alias="authentication"
+    )
+    parameters: Optional[Json] = Field(..., alias="parameters")
+    secret_parameters: Optional[list[SaasSecretParametersV1]] = Field(
+        ..., alias="secretParameters"
+    )
+    resource_templates: list[SaasResourceTemplate] = Field(
+        ..., alias="resourceTemplates"
+    )
+    self_service_roles: Optional[list[RoleV1]] = Field(..., alias="selfServiceRoles")
```

Comment:
I feel you; this was my first thought as well. Here are my thoughts on it:
* I don't know where to put these models other than here
* assuming the return value of `get_saas_files` should be independent of an integration would mean I need another layer for all "saasherder" integrations. Maybe it's worth having a dedicated module for all saasherder integration?
* I can't put it into `utils/saasherder/models.py` because these models must be independent of the GQL schema.